src/compass/utils/lut.py

```diff
@@ -117,4 +132,202 @@ def compute_geocoding_correction_luts(burst, dem_path,
+    lon_radar_grid = open_raster(f'{output_path}/x.rdr')
+
+    # Use scipy RGI to resample SET from geocoded to radar coordinates
+    pts_src = (np.flipud(lat_geo_array), lon_geo_array)
+    pts_dst = (lat_radar_grid.flatten(), lon_radar_grid.flatten())
+
+    rdr_set_e = resample_set(set_e, pts_src, pts_dst).reshape(
+        lat_radar_grid.shape)
+    rdr_set_n = resample_set(set_n, pts_src, pts_dst).reshape(
+        lat_radar_grid.shape)
+    rdr_set_u = resample_set(set_u, pts_src, pts_dst).reshape(
+        lat_radar_grid.shape)
+
+    # Convert SET from ENU to range/azimuth coordinates
+    # Note: rdr2geo heading angle is measured wrt to the East and it is positive
+    # anti-clockwise. To convert ENU to LOS, we need the azimuth angle which is
+    # measured from the north and positive anti-clockwise
+    # azimuth_angle = heading + 90
+    inc_angle = open_raster(f'{output_path}/incidence_angle.rdr')
+    head_angle = open_raster(f'{output_path}/heading_angle.rdr')
+    set_rg = enu2los(rdr_set_e, rdr_set_n, rdr_set_u, inc_angle,
+                     az_angle=head_angle + 90.0)
+    set_az = en2az(rdr_set_e, rdr_set_n, head_angle + 90.0)
+
+    return set_rg, set_az
+
+
+def compute_rdr2geo_rasters(burst, ellipsoid, dem_raster, output_path):
+    '''
+    Get latitude, longitude, incidence and
+    azimuth angle on multi-looked radar grid
+
+    Parameters
+    ----------
+    burst: Sentinel1Slc
+        S1-A/B burst object
+    ellipsoid: isce3.ext.isce3.core
+        ISCE3 Ellipsoid object
+    dem_raster: isce3.io.Raster
+        ISCE3 object including DEM raster
+    output_path: str
+        Path where to save output rasters
+    '''
+
+    # Get radar and doppler grid
+    rdr_grid = burst.as_isce3_radargrid()
+    coarse_rdr_grid = rdr_grid.multilook(64, 429)
+    grid_doppler = isce3.core.LUT2d()
+
+    # Initialize the rdr2geo object
+    rdr2geo_obj = isce3.geometry.Rdr2Geo(coarse_rdr_grid, burst.orbit,
+                                         ellipsoid, grid_doppler,
+                                         threshold=1.0e8)
+
+    # Get the rdr2geo raster needed for SET computation
+    topo_output = {'x': (True, gdal.GDT_Float64),
+                   'y': (True, gdal.GDT_Float64),
+                   'incidence_angle': (True, gdal.GDT_Float32),
+                   'heading_angle': (True, gdal.GDT_Float32)}
+    raster_list = [
+        isce3.io.Raster(f'{output_path}/{fname}.rdr', coarse_rdr_grid.width,
+                        coarse_rdr_grid.length, 1, dtype, 'ENVI')
+        if enabled else None
+        for fname, (enabled, dtype) in topo_output.items()]
+    x_raster, y_raster, incidence_raster, heading_raster = raster_list
+
+    # Run rdr2geo on coarse radar grid
+    rdr2geo_obj.topo(dem_raster, x_raster, y_raster,
+                     incidence_angle_raster=incidence_raster,
+                     heading_angle_raster=heading_raster)
+
+
+def open_raster(filename, band=1):
```

Comment:
Seems like this might be useful in `utils/helpers.py`, since a quick search shows it might be useful in 5 other places:

```
src/compass/utils/elevation_antenna_pattern.py:    slc_in = gdal.Open(path_slc_vrt, gdal.GA_ReadOnly)
src/compass/utils/helpers.py:        gdal.Open(dem_path, gdal.GA_ReadOnly)
src/compass/utils/lut.py:    ds = gdal.Open(filename, gdal.GA_ReadOnly)
src/compass/utils/range_split_spectrum.py:    in_ds = gdal.Open(burst_path, gdal.GA_ReadOnly)
src/compass/utils/raster_polygon.py:        ds = gdal.Open(dataset_path, gdal.GA_ReadOnly)
src/compass/utils/validate_cslc.py:    dataset = gdal.Open(h5_gdal_path, gdal.GA_ReadOnly)
```

Reply:
I have moved the function in `helpers.py`. However, for some of the functions you have listed , we just read the geotranformation. So I have not recycled `open_raster` for that cases.